### DISCUSSION TITLE
GEN_ENS_PROD failed to produce output files for RRFS ensemble

### DISCUSSION THREAD

**Discussion by BinbinZhou-NOAA** [dtcenter/METplus#2435] [Configuration] [answered]
I tried to run GEN_ENS_PROD tool in METplus to produce RRFS snowfall ensemble. But it failed.
There is no any  ERROR or WARNING  message in the log file.  
The GEN_ENS_PROD_OUTPUT_DIR defined in the conf file is also not generated 
But similar run for HREF is working. 
A completed simple testing case (only for one forecast hour) can be found  in the directory
    /lfs/h2/emc/vpppg/noscrub/binbin.zhou/met/RRFS on WCOSS2/Cactus.
The testing scripts are run_href_snow6h.sh and  run_rrfs_snow6h.sh for href and rrfs, respectively

Thanks!

Binbin

NCEP/EMC

**Comment by JohnHalleyGotway** [comment, marked as answer]
@BinbinZhou-NOAA, thanks to @georgemccabe for hopping on a call and debugging this with me on Hera. You should make 2 important changes to your `GenEnsProd_rrfs.conf` file.

Comment out, delete entirely, or actually define the following line:
```
# LOG_LINE_FORMAT =
```
This is causing all the blank lines you're probably seeing in your METplus log output file. Setting the format as *blank* causes empty lines to be written. Commenting it out enables the warning/error messages to actually be written using the default log line formatting. And with them being printed, you'll be able to see the following error message:
```
11/20 21:20:33.277 metplus (command_builder.py:257) ERROR: (command_builder.py:862) Found more files than expected! Found 14 expected 10. Adjust wildcard expression in template or adjust number of expected members (N_MEMBERS). 
```
The RRFS configuration tells METplus to expect 10 members, but it finds 14! Just update the config to set it to 14 instead:
```
GEN_ENS_PROD_N_MEMBERS = 14
```
After making those 2 changes, I was able to get a successful run.

@georgemccabe, based on this, we could consider enhancing METplus to do some validation of the `LOG_LINE_FORMAT` and bail out if its not set in a sensible way.

Hopefully, this helps get you rolling @BinbinZhou-NOAA.

**Comment by BinbinZhou-NOAA** [comment]
John,

  Yes, after the 2 modifications, it works now.

Thanks!

Binbin
